Quick Start -> Installing Prometheus Operator (#87)
prometheus-operator/website: content/docs/getting-started/installation.md

---
title: "Installing Prometheus Operator"
description: "Installation guide listing all the installation methods of Prometheus Operator."
date: 2020-11-16T13:59:39+01:00
draft: false
images: []
menu:
  docs:
    parent: "getting-started"
weight: 102
toc: true
---

> We offer few approaches to install Prometheus Operator in your Kubernetes cluster:

Reviewer: ```suggestion
There are different approaches to install Prometheus Operator in your Kubernetes cluster.
```

I think it makes less personal remove `We offer` since it's a offer provided by the project itself.

Wdyt?

Author: Yeah you are right.

- [Install using YAML files](#install-using-yaml-files)
- [Install using Kube-Prometheus](#install-using-kube-prometheus)

### Pre-requisites

> For all the approaches that are listed on this page, all you will need is access to a **Kubernetes cluster with kubectl!!** For this, you can check the official docs of Kubernetes [here](https://kubernetes.io/docs/tasks/tools/)

Reviewer: ```suggestion
For all the approaches listed on this page, you require access to a **Kubernetes cluster!** For this, you can check the official docs of Kubernetes available [here](https://kubernetes.io/docs/tasks/tools/).
```

```shell
# Create the namespace and CRDs, and then wait for them to be available before creating the remaining resources
kubectl create -f manifests/setup

# Wait until the "servicemonitors" CRD is created. The message "No resources found" means success in this context.
until kubectl get servicemonitors --all-namespaces ; do date; sleep 1; echo ""; done

kubectl create -f manifests/
```

We create the namespace and CustomResourceDefinitions first to avoid race conditions when deploying the monitoring components.
Alternatively, the resources in both folders can be applied with a single command:

```
kubectl create -f manifests/setup -f manifests
```

But it may be necessary to run the command multiple times for all components to be created successfully.

> Note: For versions before Kubernetes v1.20.z refer to the [Kubernetes compatibility matrix](https://github.com/prometheus-operator/kube-prometheus#kubernetes-compatibility-matrix) in order to choose a compatible branch.

> Note: If you used Kube-Prometheus as the installation method, we would recommend you to follow this page to learn how to access the resources provided.

> {{<alert icon="👉" text="If you used Kube-Prometheus as the installation method, we would recommend you to follow this page to learn how to access the resources provided." />}}

Author: There is info on how to access Prometheus, Alertmanager and Grafana on https://prometheus-operator.dev/docs/getting-started/quick-start/ . First, I will create a new page in the Kube-Prometheus guide and then mention the link to that page here.

For this, I think the new page should be created in the kube-prometheus repo and then pulled here. Wdyt?

Reviewer 2: > There is info on how to access Prometheus, Alertmanager and Grafana on https://prometheus-operator.dev/docs/getting-started/quick-start/ . First, I will create a new page in the Kube-Prometheus guide and then mention the link to that page here.
> 
> For this, I think the new page should be created in the kube-prometheus repo and then pulled here. Wdyt?

Yes, sounds correct! Please ping us in slack once you open the PR, I think @nicolastakashi and I don't watch that repository that often

Author: After the [PR](https://github.com/prometheus-operator/kube-prometheus/pull/2455) gets merged, I will add the link for this.
